google/cloud/internal/filesystem.cc

```diff
@@ -170,6 +170,26 @@ std::uintmax_t file_size(std::string const& path,
   return static_cast<std::uintmax_t>(stat.st_size);
 }
 
+std::string PathAppend(std::string const& directory, std::string const& path) {
+#if _WIN32
+  auto constexpr kSeparator = '\\';
```

Comment:
Observation: The Windows APIs allow `'/'` as a separator, so the conditional compilation isn't really necessary.

Reply:
Yes, but I do not think we want `PathAppend("foo\\", "bar")` to return `"foo\\/bar"?